### PR TITLE
fix: 3.0 Client cannot connect to server on subdirectory

### LINKED ISSUE #1775: 3.0 Client cannot connect to server on subdirectory
##  My Setup

- Operating System: Windows 10, Mac macOS 10.15
- App Version: 3.0.0, 3.0.1
- Installation type: exe

- [x] I have tested with the latest version
- [x] I can simulate the issue easily

## Description

I'm running my RC server with a subdirectory.
When I run Add Server on version 3.0, the app tries to access the URL where the subdirectory was lost.

### Current Behavior
I enter a URL containing a subdirectory in Add Server (e.g., `https://www.example.com/chat/`), it will try to open the page with the subdirectory part removed, as in `https://www.example.com/`.

### Expected Behavior
I enter a URL containing a subdirectory in Add Server (e.g., `https://www.example.com/chat/`), it will open the page as is.
Version 2.17.11 works perfectry.

### PR DESCRIPTION
**Pull request by ura14h**
Closes #1775

### CODE REVIEW

**Review by tassoevan — CHANGES_REQUESTED** [review]
Unfortunately, we need to get that resolved URL for instances reached by a URL that replies a HTTP 3xx response. I guess it should be fix at the `fetchServerInformation()` function itself.

**Comment by ura14h** [comment]
Thank you for pointing out. I have confirmed that the additional commit by tassoevan works on my server.

File: src/servers/main.ts [hunk 1]
```diff
@@ -80,7 +80,9 @@ const fetchServerInformation = async (url: URL): Promise<[finalURL: URL, version
     throw new Error();
   }
 
-  return [new URL('/', convertToURL(response.url)), responseBody.version];
+  const finalEndpoint = convertToURL(response.url);
+
+  return [new URL('../..', finalEndpoint), responseBody.version];
```
Comment (ura14h) [inline src/servers/main.ts:85]: @tassoevan 
Excuse me for pointing out late, maybe there are too many `..` ?

Comment (tassoevan) [reply src/servers/main.ts:85]: Well, the URL for reference is `<server-root>/api/info`.

Comment (ura14h) [reply src/servers/main.ts:85]: I tested another server url such `https://foo.bar/apps/chat/`, then the result is `https://foo.bar/apps` which losts `/chat`.

**Comment by julijane** [comment]
@tassoevan I'm confused here. Is this supposed to be fixed in 3.0.3? Because we have the issue with 3.0.3 still.

File: src/servers/main.ts (hunk 1 above)
Comment (huhmare) [reply src/servers/main.ts:85]: Can confirm:

Windows Rocket.Chat.Electron 3.0.3, rocketchat URL: 'http://foo.bar/rocketchat'. DevTool shows that it makes successfull query to 'http://foo.bar/rocketchat/api/info' (HTTP 200, {version: "3.1.1", success: true}).

`console.log((new URL('../..', 'http://foo.bar/rocketchat/api/info')).href); // -> 'https://foo.bar/'`

Comment (tassoevan) [reply src/servers/main.ts:85]: Oh man, you guys are right. I forgot that previously I was traversing a URL with the trailing slash. Please wait, I'll try to add some tests to assert this kind of stuff.

**Comment by ura14h** [comment]
The same issue is still open. #1796

**Pull request merged**